File: array/66-Plus-One.js
/*
66. Plus One
https://leetcode.com/problems/plus-one/
21 June 2023
David Pratt
*/

/**
 * @param {number[]} digits
 * @return {number[]}
 */

var plusOne = function(digits) {
   let str = ""
   
   // turn digits into a string
   for(let i = 0; i < digits.length; i += 1){
       str += digits[i]
   }

   // Handle large integers
   let numString = BigInt(str) + BigInt(1)

   let finalStr = String(numString)
   let finalArr = []

   // load digits into an array 
   for(let i = 0; i < finalStr.length; i += 1){
       finalArr.push(Number(finalStr[i]))
   }

   return finalArr;
};

File: string/66-Plus-One.py
/*
66. Plus One
https://leetcode.com/problems/plus-one/
19 November 2023
David Pratt

*/

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        string = ""

        for i in range(len(digits)):
           string += str(digits[i])
        new_num = str( int(string) + 1 )
        arr = []

        for i in range(len(new_num)):
           arr.append( int(new_num[i]) )

        return arr
